packages/hardhat-core/src/internal/cli/autocomplete.ts

```diff
@@ -125,39 +176,67 @@ export async function complete({
     }
 
     const isTaskParam =
-      task !== undefined &&
-      tasks[task]?.paramDefinitions[paramName]?.isFlag === false;
+      taskDefinition?.paramDefinitions[paramName]?.isFlag === false;
 
     if (isTaskParam) {
       return HARDHAT_COMPLETE_FILES;
     }
   }
 
-  // if there's no task, we complete either tasks or params
-  if (task === undefined || tasks[task] === undefined) {
+  // If there's no task or scope, we complete either tasks and scopes or params
+  if (
+    (taskName === undefined || tasks[taskName] === undefined) &&
+    (scopeName === undefined || scopes[scopeName] === undefined)
```

Comment:
Can this condition be simplified to `taskDefinition === undefined && scopeDefinition === undefined)`?